plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,412 @@
+<?php
+
+namespace Automattic\WooCommerce\DataBase\Migrations\CustomOrderTable;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer;
+use WP_CLI;
+
+/**
+ * CLI tool for migrating order data to/from custom table.
+ *
+ * Credits https://github.com/liquidweb/woocommerce-custom-orders-table/blob/develop/includes/class-woocommerce-custom-orders-table-cli.php.
+ *
+ * Class CLIRunner
+ */
+class CLIRunner {
+
+	/**
+	 * CustomOrdersTableController instance.
+	 *
+	 * @var CustomOrdersTableController
+	 */
+	private $controller;
+
+	/**
+	 * DataSynchronizer instance.
+	 *
+	 * @var DataSynchronizer;
+	 */
+	private $synchronizer;
+
+	/**
+	 * PostsToOrdersMigrationController instance.
+	 *
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $post_to_cot_migrator;
+
+	/**
+	 * Init method, invoked by DI container.
+	 *
+	 * @param CustomOrdersTableController      $controller Instance.
+	 * @param DataSynchronizer                 $synchronizer Instance.
+	 * @param PostsToOrdersMigrationController $posts_to_orders_migration_controller Instance.
+	 *
+	 * @internal
+	 */
+	final public function init( CustomOrdersTableController $controller, DataSynchronizer $synchronizer, PostsToOrdersMigrationController $posts_to_orders_migration_controller ) {
+		$this->controller           = $controller;
+		$this->synchronizer         = $synchronizer;
+		$this->post_to_cot_migrator = $posts_to_orders_migration_controller;
+	}
+
+	/**
+	 * Registers commands for CLI.
+	 */
+	public function register_commands() {
+		WP_CLI::add_command( 'wc cot count', array( $this, 'count' ) );
+		WP_CLI::add_command( 'wc cot migrate', array( $this, 'migrate' ) );
+		WP_CLI::add_command( 'wc cot verify_cot_data', array( $this, 'verify_cot_data' ) );
+	}
+
+	/**
+	 * Check if the COT feature is enabled.
+	 *
+	 * @param bool $log Optionally log a error message.
+	 *
+	 * @return bool Whether the COT feature is enabled.
+	 */
+	private function is_enabled( $log = true ) {
```

Comment:
Minor and non-essential, but here and elsewhere in the change, since these are new methods I wonder if we could add return types (at least in simple cases where the return type is not mixed). Ex:

```suggestion
	private function is_enabled( $log = true ): bool {
```

Reply:
thanks, added in https://github.com/woocommerce/woocommerce/pull/32938/commits/75425a8cc878aa1bfe785d6b1cf9321c37a544d0